xdsl/irdl/operations.py

```diff
@@ -1356,17 +1356,12 @@ def get_variadic_sizes_from_attr(
             f"Expected {size_attribute_name} {container_name} in {op.name} operation."
         )
     attribute = container[size_attribute_name]
-    if not isa(attribute, DenseArrayBase):
+    if not isa(attribute, DenseArrayBase[I32]):
         raise VerifyException(
             f"{size_attribute_name} {container_name} is expected "
-            "to be a DenseArrayBase."
+            "to be a DenseArrayBase of i32."
         )
 
-    if attribute.elt_type != i32:
-        raise VerifyException(
-            f"{size_attribute_name} {container_name} is expected to "
-            "be a DenseArrayBase of i32"
-        )
     def_sizes = cast(Sequence[int], attribute.get_values())
```

Comment:
Unnecessary cast

Reply:
So my Pyright didn't catch this as unnecessary for some reason, I'll look into it

Reply:
I think `tuple[int, ...]` -> `Sequence[int]` is technically doing something, but it's not needed here